glustercli/cmd/root.go

```diff
@@ -17,7 +19,28 @@ var RootCmd = &cobra.Command{
 		if err != nil {
 			fmt.Println("Error initializing log file ", err)
 		}
-		initRESTClient(flagEndpoints[0], flagUser, flagSecret, flagCacert, flagInsecure)
+
+		if flagAuthFile == "" && flagSecret == "" {
+			data, err := ioutil.ReadFile(defaultAuthPath)
+			if err != nil && !os.IsNotExist(err) {
+				failure("failed to read secret", err, 1)
```

Comment:
Do not fail here(you can add verbose log), try without secret since CLI doesn't know that restauth is enabled or not. set secret to empty string.

Reply:
This can also happen when glustercli is run from outside the cluster(Example: Run glustercli from laptop to manage remote Gluster clusters)

Reply:
done